app/controllers/product_access_groups_controller.rb

```diff
@@ -2,6 +2,7 @@ class ProductAccessGroupsController < ApplicationController
 
   before_action :authenticate_user!
   before_action :init_current_facility
+  before_action :init_current_product
```

Comment:
Pretty sure this needs to happen after the `load_and_authorize_resource :instrument`, otherwise `@instrument isn't set yet.